tests/validation_tests/round_four_specific_validations/test_tf_r4_specific_validations.py

```diff
@@ -421,3 +382,110 @@ def test_validate_psi_funding_gap_success():
     failures = validate_psi_funding_gap(workbook)
 
     assert failures is None
+
+
+def test_validate_locations_success():
+    # validate 3 projects
+    project_details_df = pd.DataFrame(
+        data=[
+            # Project 1: Single - valid Locations and Lat/Long data
+            {
+                "Single or Multiple Locations": "Multiple",
+                "GIS Provided": YesNoEnum.YES,
+                "Locations": "AB1 2CD",
+                "Lat/Long": "Test Coords",
+            },
+            # Project 2: Multiple - valid Locations, Lat/Long data and GIS Provided data
+            {
+                "Single or Multiple Locations": "Single",
+                "GIS Provided": pd.NA,
+                "Locations": "AB1 2CD",
+                "Lat/Long": "Test Coords",
+            },
+            # Project 3: Invalid Single / Multiple - invalid Locations, Lat/Long data and GIS Provided data
+            # this should not produce failures because the Single / Multiple value is invalid (this is picked up during
+            # schema validation)
+            {"Single or Multiple Locations": "Invalid Value", "GIS Provided": pd.NA, "Locations": "", "Lat/Long": ""},
+        ]
+    )
+    workbook = {"Project Details": project_details_df}
+
+    failures = validate_locations(workbook)
+
+    assert failures == []
```

Comment:
Why doesn't this produce a failure like: "There are blank cells in column: Single or Multiple Locations Use the space provided to tell us the relevant information"

Reply:
That case is covered by the validation schema enum validation as usual.

Reply:
Ah, okay. Got it this is a seperate validation